src/support/CMakeLists.txt

```diff
@@ -14,4 +14,15 @@ set(support_SOURCES
   utilities.cpp
   ${support_HEADERS}
 )
+
+#[[ The below condition is intended for removal once the suffix_tree and suffix_tree_node source files no longer depend on LLVM code in the third_party folder ]]
+if(EMSCRIPTEN)
 add_library(support OBJECT ${support_SOURCES})
+else()
+set(support_with_suffix_tree_SOURCES
+  suffix_tree.cpp
+  suffix_tree_node.cpp
+  ${support_SOURCES}
+)
+add_library(support OBJECT ${support_with_suffix_tree_SOURCES})
```

Comment:
I _think_ that we usually have two spaces of indent for code inside the if-else.

Reply:
done